client/client.go

```diff
@@ -130,50 +115,53 @@ func makeClient(cfg clientConfig) (Client, error) {
 }
 
 type clientConfig struct {
-	// URLs when specified will create an HTTP client.
-	urls []string
-	// Insecure will allow creating the HTTP client without a bound group.
-	insecure bool
+	// clients is the set of options for fetching randomness
+	clients []Client
+	// watcher is a constructor function for generating a new partial client of randomness
+	watcher WatcherCtor
 	// from `chainInfo.Hash()` - serves as a root of trust for a given
 	// randomness chain.
 	chainHash []byte
 	// Full chain information - serves as a root of trust.
 	chainInfo *chain.Info
-	// transport configures the http parameters used when fetching randomness.
-	transport http.RoundTripper
+	// insecure indicates the root of trust does not need to be present.
+	insecure bool
 	// cache size - how large of a cache to keep locally.
 	cacheSize int
 	// customized client log.
 	log log.Logger
 	// time after which a watcher will failover to using client.Get to get the latest randomness.
 	failoverGracePeriod time.Duration
-	// watcher is a constructor function that creates a new Watcher
-	watcher WatcherCtor
 	// autoWatch causes the client to start watching immediately in the background so that new randomness is proactively fetched and added to the cache.
 	autoWatch bool
 	// prometheus is an interface to a Prometheus system
 	prometheus prometheus.Registerer
 }
 
+func (c *clientConfig) tryPopulateInfo(cli Client) {
+	if c.chainInfo == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+		c.chainInfo, _ = cli.Info(ctx)
```

Comment:
What if it fails ?  Shouldn't it return an error propagated through the construction of the `makeClient` ?